server/models/AccountingCategory.ts

```diff
@@ -0,0 +1,134 @@
+import type {
+  BelongsToGetAssociationMixin,
+  ForeignKey,
+  HasManyGetAssociationsMixin,
+  InferAttributes,
+  InferCreationAttributes,
+  NonAttribute,
+} from 'sequelize';
+
+import ActivityTypes from '../constants/activities';
+import sequelize, { DataTypes, Model } from '../lib/sequelize';
+
+import Collective from './Collective';
+import User from './User';
+import models, { Expense } from '.';
+
+type AccountingCategoryCreationAttributes = InferCreationAttributes<
+  AccountingCategory,
+  { omit: 'id' | 'createdAt' | 'updatedAt' }
+>;
+
+type AccountingCategoryEditActivityData = {
+  added?: Array<Partial<AccountingCategory>>;
+  removed?: Array<Partial<AccountingCategory>>;
+  edited?: Array<{ previousData: Partial<AccountingCategory>; newData: Partial<AccountingCategory> }>;
+};
+
+class AccountingCategory extends Model<InferAttributes<AccountingCategory>, AccountingCategoryCreationAttributes> {
+  declare id: number;
+  declare CollectiveId: ForeignKey<Collective['id']>;
+  declare code: string;
+  declare name: string;
+  declare friendlyName?: string;
+  declare createdAt: Date;
+  declare updatedAt: Date;
+
+  // Associations
+  declare getCollective: BelongsToGetAssociationMixin<Collective>;
+  declare getExpenses: HasManyGetAssociationsMixin<Expense>;
+
+  declare collective?: Collective;
+  declare expenses?: Expense[];
+
+  // Static methods
+  public static async createEditActivity(
+    collective: Collective,
+    user: User,
+    data: AccountingCategoryEditActivityData,
+  ): Promise<void> {
+    await models.Activity.create({
+      type: ActivityTypes.ACCOUNTING_CATEGORIES_EDITED,
+      UserId: user.id,
+      CollectiveId: collective.id,
+      HostCollectiveId: collective.HostCollectiveId,
+      data,
+    });
+  }
```

Comment:
Not sure we need this shim, especially if we're only calling it from the mutation.

Reply:
This was initially done to have a common helper between create/edit/remove mutations, but it's indeed not helpful anymore.

Reply:
Actually, since we don't yet have good typing for activitie's data, this helper helps to plug typescript on it. I'll therefore keep it for now